packages/patternfly-4/react-core/src/components/Modal/Modal.tsx

```diff
@@ -1,76 +1,80 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import PropTypes from 'prop-types';
-import ModalContent from './ModalContent';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
 import { canUseDOM } from 'exenv';
-import { KEY_CODES } from '../../helpers/constants';
+
 import { css } from '@patternfly/react-styles';
 import styles from '@patternfly/patternfly/components/Backdrop/backdrop.css';
 
-const propTypes = {
+import { KEY_CODES } from '../../helpers/constants';
+import ModalContent from './ModalContent';
+
+export interface ModalProps extends React.HTMLProps<HTMLDivElement> {
   /** content rendered inside the Modal. */
-  children: PropTypes.node.isRequired,
+  children: React.ReactNode;
   /** additional classes added to the Modal */
-  className: PropTypes.string,
+  className?: string;
   /** Flag to show the modal */
-  isOpen: PropTypes.bool,
+  isOpen?: boolean;
   /** Content of the Modal Header */
-  title: PropTypes.string.isRequired,
-  /** Flag to show the title */
-  hideTitle: PropTypes.bool,
+  title: string;
+  /** Flag to hide the title */
+  hideTitle?: boolean;
   /** id to use for Modal Box description */
-  ariaDescribedById: PropTypes.string,
+  ariaDescribedById?: string;
   /** Action buttons to put in the Modal Footer */
-  actions: PropTypes.any,
+  actions?: any,
   /** A callback for when the close button is clicked */
-  onClose: PropTypes.func,
+  onClose?(): void;
```

Comment:
Nitpicky, but we've been doing `onClose?: () => void`. Sometimes different docgen versions pick it up differently.